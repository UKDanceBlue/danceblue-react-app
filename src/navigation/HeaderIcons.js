// Import third-party dependencies
import React from 'react';
import { View, Pressable, PixelRatio } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome5';

const HeaderIcons = ({ navigation, color }) => (
  <View
    style={[
      {
        flexDirection: 'row',
      },
    ]}
  >
    <Pressable onPress={() => navigation.navigate('Profile')}>
      <Icon
        name="bell"
        color={color}
        style={{ textAlignVertical: 'center', padding: '5%', fontSize: PixelRatio.get() * 8 }}
      />
    </Pressable>
    <Pressable onPress={() => navigation.navigate('Profile')}>
      <Icon
        name="user"
        color={color}
        style={{ textAlignVertical: 'center', padding: '5%', fontSize: PixelRatio.get() * 8 }}
      />
    </Pressable>
  </View>
);

export default HeaderIcons;
