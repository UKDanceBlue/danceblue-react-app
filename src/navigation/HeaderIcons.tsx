import { FontAwesome5 } from "@expo/vector-icons";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { View } from "native-base";
import { PixelRatio, TouchableOpacity } from "react-native";

import { RootStackParamList } from "../types/navigationTypes";

const HeaderIcons = ({
  navigation, color
}: {
  navigation: NativeStackNavigationProp<RootStackParamList>;
  color?: string;
}) => (
  <View
    style={{ flexDirection: "row" }}
  >
    <TouchableOpacity onPress={() => navigation.navigate("Notifications")}>
      <FontAwesome5
        name="bell"
        color={color}
        style={{ textAlignVertical: "center", padding: "5%", fontSize: PixelRatio.get() * 8 }}
      />
    </TouchableOpacity>
    <TouchableOpacity onPress={() => navigation.navigate("Profile")}>
      <FontAwesome5
        name="user"
        color={color}
        style={{ textAlignVertical: "center", padding: "5%", fontSize: PixelRatio.get() * 8 }}
      />
    </TouchableOpacity>
  </View>
);

export default HeaderIcons;
