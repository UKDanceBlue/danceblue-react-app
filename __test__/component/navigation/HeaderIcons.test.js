import React from 'react';
import renderer from 'react-test-renderer';
import { loadAsync } from 'expo-font';
import { FontAwesome5 } from '@expo/vector-icons';
import HeaderIcons from '../../../src/navigation/HeaderIcons';

beforeAll(async () => loadAsync(FontAwesome5.font));

test('renders correctly', () => {
  const tree = renderer.create(<HeaderIcons />).toJSON();
  expect(tree).toMatchSnapshot();
});
