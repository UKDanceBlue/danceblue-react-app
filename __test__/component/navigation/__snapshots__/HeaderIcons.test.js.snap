// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "fontSize": 16,
            "padding": "5%",
            "textAlignVertical": "center",
          },
          Object {
            "fontFamily": "FontAwesome5Free-Regular",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "400",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "fontSize": 16,
            "padding": "5%",
            "textAlignVertical": "center",
          },
          Object {
            "fontFamily": "FontAwesome5Free-Regular",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "400",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
